src/macaron/policy_engine/souffle.py

```diff
@@ -0,0 +1,177 @@
+# Copyright (c) 2023 - 2023, Oracle and/or its affiliates. All rights reserved.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/.
+
+"""
+Wrapper classes for invoking souffle by subprocess and getting the resulting tables.
+
+Implements a context manager to create and clean up temporary directories.
+"""
+
+import csv
+import glob
+import logging
+import os
+import shutil
+import subprocess  # nosec B404
+import tempfile
+from types import TracebackType
+from typing import Optional
+
+logger: logging.Logger = logging.getLogger(__name__)
+
+
+class SouffleError(Exception):
```

Comment:
Similar comment as other errors. Please add the discussed TODO.

Reply:
Fixed in https://github.com/oracle-samples/macaron/commit/7584d191a925261dd109b81512af671aa81d42aa